examples/subscriptions/app/routes/products.$handle.tsx

```diff
@@ -0,0 +1,505 @@
+  variants: Promise<ProductVariantsQuery>;
+}) {
+  const {title, descriptionHtml, sellingPlanGroups} = product;
+  return (
+    <div className="product-main">
+      <h1>{title}</h1>
+      <ProductPrice selectedVariant={selectedVariant} />
+      <br />
+      <Suspense
+        fallback={
+          <ProductForm
+            product={product}
+            selectedVariant={selectedVariant}
+            variants={[]}
+            sellingPlanGroups={sellingPlanGroups}
+          />
+        }
+      >
+        <Await
+          errorElement="There was a problem loading product variants"
+          resolve={variants}
+        >
+          {(data) => (
+            <ProductForm
+              product={product}
+              selectedVariant={selectedVariant}
+              variants={data.product?.variants.nodes || []}
+              sellingPlanGroups={sellingPlanGroups}
+            />
+          )}
+        </Await>
+      </Suspense>
+      <br />
+      <br />
+      <p>
+        <strong>Description</strong>
+      </p>
+      <br />
+      <div dangerouslySetInnerHTML={{__html: descriptionHtml}} />
+      <br />
+    </div>
+  );
+}
+
+function ProductPrice({
+  selectedVariant,
+}: {
+  selectedVariant: ProductFragment['selectedVariant'];
+}) {
+  return (
+    <div className="product-price">
+      {selectedVariant?.compareAtPrice ? (
+        <>
+          <p>Sale</p>
+          <br />
+          <div className="product-price-on-sale">
+            {selectedVariant ? <Money data={selectedVariant.price} /> : null}
+            <s>
+              <Money data={selectedVariant.compareAtPrice} />
+            </s>
+          </div>
+        </>
+      ) : (
+        selectedVariant?.price && <Money data={selectedVariant?.price} />
+      )}
+    </div>
+  );
+}
+
+function ProductForm({
+  product,
+  selectedVariant,
+  variants,
+  sellingPlanGroups,
+}: {
+  product: ProductFragment;
+  selectedVariant: ProductFragment['selectedVariant'];
+  variants: Array<ProductVariantFragment>;
+  sellingPlanGroups: ProductFragment['sellingPlanGroups'];
+}) {
+  const {pathname, search} = useLocation();
+  const linkParams = new URLSearchParams(search);
+  const selectedSellingPlan = linkParams.get('selling_plan');
+  return (
+    <div className="product-form">
+      <VariantSelector
+        handle={product.handle}
+        options={product.options}
+        variants={variants}
+      >
+        {({option}) => <ProductOptions key={option.name} option={option} />}
+      </VariantSelector>
+
+      {sellingPlanGroups?.nodes && (
+        <div>
+          {sellingPlanGroups.nodes.map((sellingPlanGroup) => {
+            const {name, sellingPlans} = sellingPlanGroup;
+
+            return (
+              <>
+                <p className="mb-2">
+                  <strong>{name}:</strong>
+                </p>
+                {sellingPlans.nodes.map((sellingPlan) => {
+                  linkParams.set('selling_plan', sellingPlan.id);
```

Comment:
Thoughts on storing the entire GID in the url? Kind of ugly, but simple.

Reply:
changed it to just include the id